lib/web/fetch/index.js

```diff
@@ -133,116 +178,147 @@ function fetch (input, init = undefined) {
 
-  // 10. Let controller be null.
-  let controller = null
+    // 8. Let relevantRealm be this’s relevant Realm.
+    const relevantRealm = null
 
-  // 11. Add the following abort steps to requestObject’s signal:
-  addAbortListener(
-    requestObject.signal,
-    () => {
-      // 1. Set locallyAborted to true.
-      locallyAborted = true
+    // 9. Let locallyAborted be false.
+    let locallyAborted = false
 
-      // 2. Assert: controller is non-null.
-      assert(controller != null)
+    // 10. Let controller be null.
+    let controller = null
 
-      // 3. Abort controller with requestObject’s signal’s abort reason.
-      controller.abort(requestObject.signal.reason)
+    // 11. Add the following abort steps to requestObject’s signal:
+    addAbortListener(
+      requestObject.signal,
+      () => {
+        // 1. Set locallyAborted to true.
+        locallyAborted = true
 
-      // 4. Abort the fetch() call with p, request, responseObject,
-      //    and requestObject’s signal’s abort reason.
-      abortFetch(p, request, responseObject, requestObject.signal.reason)
-    }
-  )
+        // 2. Assert: controller is non-null.
+        assert(controller != null)
 
-  // 12. Let handleFetchDone given response response be to finalize and
-  // report timing with response, globalObject, and "fetch".
-  const handleFetchDone = (response) =>
-    finalizeAndReportTiming(response, 'fetch')
+        // 3. Abort controller with requestObject’s signal’s abort reason.
+        controller.abort(requestObject.signal.reason)
 
-  // 13. Set controller to the result of calling fetch given request,
-  // with processResponseEndOfBody set to handleFetchDone, and processResponse
-  // given response being these substeps:
+        // 4. Abort the fetch() call with p, request, responseObject,
+        //    and requestObject’s signal’s abort reason.
+        abortFetch(p, request, responseObject, requestObject.signal.reason, context)
+      }
+    )
 
-  const processResponse = (response) => {
-    // 1. If locallyAborted is true, terminate these substeps.
-    if (locallyAborted) {
-      return
-    }
+    // 12. Let handleFetchDone given response response be to finalize and
+    // report timing with response, globalObject, and "fetch".
+    const handleFetchDone = (response) =>
+      finalizeAndReportTiming(response, 'fetch')
 
-    // 2. If response’s aborted flag is set, then:
-    if (response.aborted) {
-      // 1. Let deserializedError be the result of deserialize a serialized
-      //    abort reason given controller’s serialized abort reason and
-      //    relevantRealm.
+    // 13. Set controller to the result of calling fetch given request,
+    // with processResponseEndOfBody set to handleFetchDone, and processResponse
+    // given response being these substeps:
 
-      // 2. Abort the fetch() call with p, request, responseObject, and
-      //    deserializedError.
+    const processResponse = (response) => {
```

Comment:
If you use `handlePromise` on the final promise in 14 you can revert _all_ the changes to `processResponse` as it will capture all the resolves and rejects that come out of here anyway.